pulp_ansible/app/galaxy/v3/views.py

```diff
@@ -153,8 +153,23 @@ def get_queryset(self):
         deprecated_query = AnsibleCollectionDeprecated.objects.filter(
             collection=OuterRef("pk"), repository_version=repo_version
         )
-        collections = Collection.objects.filter(versions__in=repo_version.content).distinct()
-        collections = collections.annotate(deprecated=Exists(deprecated_query))
+        added_repo_version_query = RepositoryVersion.objects.filter(
+            repository=repo_version.repository,
+            complete=True,
+            added_memberships__content__in=OuterRef("versions__pk"),
+        ).values("pulp_last_updated")
+        removed_repo_version_query = RepositoryVersion.objects.filter(
+            repository=repo_version.repository,
+            complete=True,
+            removed_memberships__content__in=OuterRef("versions__pk"),
+        ).values("pulp_last_updated")
+
+        collections_qs_annotated = Collection.objects.annotate(
+            repo_version_added_at=Max(Subquery(added_repo_version_query)),
+            repo_version_removed_at=Max(Subquery(removed_repo_version_query)),
+            deprecated=Exists(deprecated_query),
+        )
+        collections = collections_qs_annotated.filter(versions__in=repo_version.content).distinct()
```

Comment:
When filter comes first the test fails, I believe it is because the subqueries use collection versions that are not on the current repo version.

Reply:
Yup that makes sense. Well breaking it up to not use parens I think would be better because of readability.